tests/unit/cypher_main_visitor.cpp

```diff
@@ -4088,3 +4088,88 @@ TEST_P(CypherMainVisitorTest, VersionQuery) {
   TestInvalidQuery("SHOW VERSIONS", ast_generator);
   ASSERT_NO_THROW(ast_generator.ParseQuery("SHOW VERSION"));
 }
+
+TEST_P(CypherMainVisitorTest, ForeachThrow) {
+  auto &ast_generator = *GetParam();
+  EXPECT_THROW(ast_generator.ParseQuery("FOREACH(i IN [1, 2] | "), SyntaxException);
+  EXPECT_THROW(ast_generator.ParseQuery("FOREACH(i IN [1, 2] | UNWIND"), SyntaxException);
+  EXPECT_THROW(ast_generator.ParseQuery("FOREACH(i IN [1, 2] | LOADCSV"), SyntaxException);
```

Comment:
These test cases can be improved in my opinion, because it is possible they throw because the nested statement is not valid. Therefore I would extend them to be valid: `UNWIND [1,2,3] AS j CREATE (n)`, and maybe switch `LOADCSV` to something else. If I missed the point of these tests, then adding new test cases are also fine to me.